src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs

```diff
@@ -221,6 +221,31 @@ public async Task<IReadOnlyList<JobInfo>> GetJobsByIdsAsync(byte queueType, long
             return await _sqlRetryService.ExecuteSqlDataReader(cmd, JobInfoExtensions.LoadJobInfo, _logger, "GetJobsByIdsAsync failed.", cancellationToken);
         }
 
+        public async Task<string> PeekAsync(byte queueType, CancellationToken cancellationToken)
+        {
+            try
+            {
+                using SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true);
+                using SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateRetrySqlCommand();
+                sqlCommandWrapper.CommandText = "SELECT TOP 1 JobId FROM [dbo].[JobQueue] WITH (NOLOCK) WHERE QueueType = @QueueType AND Status in (0, 1) ORDER BY JobId DESC";
+                sqlCommandWrapper.Parameters.AddWithValue("@QueueType", queueType);
+                await using SqlDataReader sqlDataReader = await sqlCommandWrapper.ExecuteReaderAsync(cancellationToken);
+
+                var result = await sqlDataReader.ReadJobInfoAsync(cancellationToken);
+                return result?.Id.ToString();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "PeekAsync failed.");
```

Comment:
Do we want to differentiate between LogError for "real" errors and LogWarning for retriable errors?

Reply:
Thanks. I updated logs to be more inline with actual errors vs warnings and informational.